airbyte-integrations/connectors/source-salesforce/source_salesforce/streams.py

```diff
@@ -633,36 +647,62 @@ def get_updated_state(self, current_stream_state: MutableMapping[str, Any], late
 
 
 class BulkIncrementalSalesforceStream(BulkSalesforceStream, IncrementalRestSalesforceStream):
+    STREAM_SLICE_STEP = 120
+
     def next_page_token(self, last_record: Mapping[str, Any]) -> Optional[Mapping[str, Any]]:
         if self.name not in UNSUPPORTED_FILTERING_STREAMS:
-            page_token: str = last_record[self.cursor_field]
-            res = {"next_token": page_token}
-            # use primary key as additional filtering param, if cursor_field is not increased from previous page
-            if self.primary_key and self.prev_start_date == page_token:
-                res["primary_key"] = last_record[self.primary_key]
-            return res
+            return {
+                "next_token": last_record[self.cursor_field],
+                "primary_key": last_record.get(self.primary_key)
+            }
         return None
 
+    def stream_slices(
```

Comment:
shouldn't we move this to the base class i.e. to implement slicing for the full refresh as well?

Reply:
Full Refresh in BulkSalesforce streams is also using this slicing. Do you think we need not only Bulk API but also REST API to use slices?

Reply:
Decided to not to move slicing for bulk full refresh stream base class

Reply:
To bring this up again, if we implement `stream_slices()` on this class `BulkIncrementalSalesforceStream`. But based on the changes to the `IncrementalRestSalesforceStream` above, I would assume that the class also need to rely on this slicing logic so we have slices w/ start/end date. Since `IncrementalRestSalesforceStream` doesn't inherit behavior from this class, is `IncrementalRestSalesforceStream.request_params()` going to work?

Reply:
IncrementalRestSalesforceStream was also updated because I wanted to have the same SQL query for both implementations. It does work the same way as before